src/model/Space.cpp

```diff
@@ -918,6 +924,121 @@ namespace model {
       return result;
     }
 
+    std::vector<Surface> Space_Impl::findSurfacesWithIncorrectOrientationRaycasting() const {
+      std::vector<Surface> result;
+
+      const Point3d p0{};
+
+      auto surfaces = this->surfaces();
+      for (const auto& surface : surfaces) {
+
+        auto outwardNormal = surface.outwardNormal();
+        auto inwardNormal = outwardNormal.reverseVector();
+
+        auto surfacePoint = surface.centroid();
+        bool found = false;
+        for (const auto& anotherSurface : surfaces) {
+          if (surface == anotherSurface) {
+            continue;
+          }
+          auto anotherPlane = anotherSurface.plane();
+          const double vd = anotherPlane.outwardNormal().dot(outwardNormal);
+
+          auto reversedPlane = anotherPlane.reversePlane();
+          const double vd2 = reversedPlane.outwardNormal().dot(inwardNormal);
+          OS_ASSERT(vd == vd2);
+
+          // If this isn't orthogonal
+          if (std::abs(vd) < 0.001) {
+            LOG(Debug, "Surface " << surface.nameString() << " is orthogonal to " << anotherSurface.nameString());
+            continue;
+          }
+          if (vd < 0) {
+            LOG(Debug, "Surfaces '" << surface.nameString() << "' and '" << anotherSurface.nameString()
+                                    << "' are seemingly pointing in the opposite outward direction, which is a good sign.");
+          } else {
+            LOG(Debug, "Surfaces '" << surface.nameString() << "' and '" << anotherSurface.nameString()
+                                    << "' are seemingly pointing in the same outward direction, potentially one is in the wrong direction.");
+          }
+
+          const double v0 = anotherPlane.outwardNormal().dot(surfacePoint - p0) + anotherPlane.d();
+          const double v02 = -(reversedPlane.outwardNormal().dot(surfacePoint - p0) + reversedPlane.d());
+          OS_ASSERT(v0 == v02);
+          const double t = v0 / vd;
+          if (t > 0.0) {
+            LOG(Debug, "Surface '" << surface.nameString() << "' raycasts to " << anotherSurface.nameString());
+            found = true;
+            break;
+          } else {
+            LOG(Debug, "Surface '" << surface.nameString() << "' does not raycast to " << anotherSurface.nameString());
+          }
+        }
+        if (!found) {
+          result.push_back(surface);
+        }
+      }
+
+      return result;
+    }
+
+    std::vector<Surface> Space_Impl::findSurfacesWithIncorrectOrientationPolyhedron(const Polyhedron& volumePoly) const {
+      auto sf3ds = volumePoly.findSurfacesWithIncorrectOrientation();
```

Comment:
This used the Polyhedron::findSurfacesWithIncorrectOrientation which checks edges. It works even if not enclosed, and not convex.